### PR TITLE
Battle HP Bar and related code

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/battle/battle.h [hunk 1]
```diff
@@ -438,59 +439,48 @@ typedef struct BattleContext {
 typedef struct BattleSystem BattleSystem;
 
 typedef struct BattleHpBar {
-    u8 script;
-    void *unk4;
-    void *unk8;
+    struct {
```
Comment (adrienntindall) [inline include/battle/battle.h:442]: why?

File: src/battle/battle_hp_bar.c [hunk 2]
```diff
@@ -0,0 +1,1756 @@
...
+
+    UnkImageStruct *ret = SpriteRenderer_LoadResourcesAndCreateSprite(renderer, gfxHandler, pRes);
+    TickSpriteAnimation1Frame(ret->sprite);
+    return ret;
+}
+
+void ov12_0226498C(BattleHpBar *hpBar, u32 num, u32 flag) {
+    GF_ASSERT(hpBar->unk4 != NULL);
+    if (hpBar->type == HP_BAR_TYPE_SAFARI) {
+        flag &= 0xC00;
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:487]: would prefer some consts here even if theyre all unown, ie #def HP_BAR_FLAG_4 (1 << 4) type things

File: src/battle/battle_hp_bar.c [hunk 3]
```diff
@@ -0,0 +1,1756 @@
...
+void BattleHpBar_LoadResources(BattleHpBar *hpBar) {
+    const UnkTemplate_0200D748 *tmplate;
+    SpriteRenderer *renderer;
+    SpriteGfxHandler *gfxHandler;
+    PaletteData *plttData;
+    NARC *narc;
+
+    narc = NARC_New(NARC_a_0_0_8, HEAP_ID_BATTLE);
+
+    renderer   = ov12_0223A8E4(hpBar->bsys);
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:634]: ov12_0223A8E4 -> BattleSystem_GetSpriteRenderer

File: src/battle/battle_hp_bar.c [hunk 4]
```diff
@@ -0,0 +1,1756 @@
...
+    gfxHandler = ov12_0223A8EC(hpBar->bsys);
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:635]: ov12_0223A8EC -> BattleSystem_GetGfxHandler

File: src/battle/battle_hp_bar.c [hunk 5]
```diff
@@ -0,0 +1,1756 @@
...
+    const UnkTemplate_0200D748 *r4;
+
+    GF_ASSERT(hpBar->unk4 != NULL);
+
+    r4 = BattleHpBar_Util_GetHpBoxSpriteTemplate(hpBar->type);
+
+    Sprite_SetPositionXY(hpBar->unk4->sprite, r4->x + x, r4->y + y);
+    if (hpBar->unk8 != NULL) {
+        Sprite_SetPositionXY(hpBar->unk8->sprite,
+            r4->x + x - ov12_0226D368[hpBar->type],
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:868]: ov12_0226D368 -> sHpBarXOffsets

Comment (adrienntindall) [reply src/battle/battle_hp_bar.c:868]: or sHpBoxXOffsets, etc, whatever you want

File: src/battle/battle_hp_bar.c [hunk 6]
```diff
@@ -0,0 +1,1756 @@
...
+    mon    = BattleSystem_GetPartyMon(hpBar->bsys, hpBar->battlerId, hpBar->monId);
+    boxMon = Mon_GetBoxMon(mon);
+    BufferBoxMonNickname(msgFormat, 0, boxMon);
+    StringExpandPlaceholders(msgFormat, string, string2);
+
+    AddTextWindowTopLeftCorner(bgConfig, &window, 8, 2, 0, 15);
+    AddTextPrinterParameterizedWithColorAndSpacing(&window, 0, string, 0, 0, TEXT_SPEED_NOTRANSFER, MAKE_TEXT_COLOR(14, 2, 15), 0, 0, NULL);
+    srcBuf = window.pixelBuffer;
+
+    {
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:1184]: brackets required to match?

**Review by adrienntindall — CHANGES_REQUESTED** [review]
Would prefer more documentation but I don't think that's feasible as of right now

File: src/battle/battle_hp_bar.c (hunk 6 above)
Comment (PikalaxALT) [reply src/battle/battle_hp_bar.c:1184]: iirc yes

File: src/battle/battle_hp_bar.c [hunk 7]
```diff
@@ -822,51 +822,51 @@ void ov12_02264E84(BattleHpBar *hpBar) {
...
+    if (hpBar->boxObj != NULL) {
+        UnkImageStruct_SetSpriteVisibleFlag(hpBar->boxObj, a1);
         ov12_02264F00(hpBar, a1);
     }
 }
 
 #ifdef NONMATCHING
 static void ov12_02264F44(BattleHpBar *hpBar, int x, int y) {
-    const UnkTemplate_0200D748 *r4;
+    const UnkTemplate_0200D748 *tmplate;
```
Comment (adrienntindall) [inline src/battle/battle_hp_bar.c:859]: tmplate -> template

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**